### ISSUE TITLE
Text too big or box too small in synthetize

### ISSUE THREAD

**Issue by RBMindee** [mindee/doctr#588]
### Bug description

Running the basic example in readme with `plt.imshow(result.synthesize()[0]); plt.axis('off'); plt.show()`, several portions of text are cut on the right, as if the text is too big or the box too small. This happens on both pdf and images.

### Code snippet to reproduce the bug

```python
from doctr.io import DocumentFile
from doctr.models import ocr_predictor
from pprint import pprint

model = ocr_predictor(pretrained=True)
doc = DocumentFile.from_pdf("path/to/pdf").as_images()
result = model(doc)
plt.imshow(result.synthesize()[0]); plt.axis('off'); plt.show()
```

### Error traceback

No error

### Environment

/home/remy/.virtualenvs/doctr_qa/lib/python3.8/site-packages/tensorflow_addons/utils/ensure_tf_install.py:53: UserWarning: Tensorflow Addons supports using Python ops for all Tensorflow versions above or equal to 2.4.0 and strictly below 2.7.0 (nightly versions are not supported). 
 The versions of TensorFlow you are currently using is 2.7.0 and is not supported. 
Some things might work, some things might not.
If you were to encounter a bug, do not file an issue.
If you want to make sure you're using a tested and supported configuration, either change the TensorFlow version or the TensorFlow Addons's version. 
You can find the compatibility matrix in TensorFlow Addon's readme:
https://github.com/tensorflow/addons
  warnings.warn(
Collecting environment information...

DocTR version: 0.4.1a0
TensorFlow version: 2.7.0
PyTorch version: N/A (torchvision N/A)
OpenCV version: 4.5.4-dev
OS: Ubuntu 20.04.3 LTS
Python version: 3.8.10
Is CUDA available (TensorFlow): Yes
Is CUDA available (PyTorch): N/A
CUDA runtime version: 11.4.120
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Laptop GPU
Nvidia driver version: 470.63.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.4

**Comment by fg-mindee** [comment]
Hi @RBMindee,
Could you provide some sample to reproduce this behaviour please? :pray:

**Comment by RBMindee** [comment]
[seatosky_20180908_005.pdf](https://github.com/mindee/doctr/files/7497376/seatosky_20180908_005.pdf)
Here's an example, see the "a" at the end of sea to sky gondola or the amount $46.15. Just in case it is related, this is with matplotlib TkAgg backend.

**Comment by fg-mindee** [comment]
@RBMindee that might be related to the matplotlib backend then :thinking: 

Here is what I get on my end:
![synthesized_page](https://user-images.githubusercontent.com/76527547/140753635-2a190c7c-6805-4098-aaa4-83eff9520630.png)

If that looks good to you, would you mind trying again with another matplotlib backend? If it's indeed related to this, we might have to throw a warning when users have the TkAgg backend

**Comment by RBMindee** [comment]
That's not what I have. What is the backend you used ?

**Comment by fg-mindee** [comment]
It's getting weird :thinking: 
```python
In [1]: import matplotlib

In [2]: matplotlib.get_backend()
Out[2]: 'TkAgg'
```

Which version of matplotlib & PIL are you running?
I have mpl 3.4.2 & PIL 8.3.1

**Comment by RBMindee** [comment]
mpl 3.4.2, PIL 8.4.0 (but I tried 8.3.1 with no change), tkinter is 8.6. What's your tkinter ?

**Comment by fg-mindee** [comment]
It's from the standard library and I have the same Python version as you :upside_down_face: 

Perhaps it's because you don't have the default font installed? When running the script, does it throw any warning about loading default PIL font?

**Comment by RBMindee** [comment]
I am not sure if it's due to my venv, but I had to install it (sudo apt install python3-tk). I do not have any warning and I have the default fonts.

**Comment by fg-mindee** [comment]
Coming back to this, we should handle this PR :+1: 

@RBMindee do you have any way to check whether it reproduces the same behaviour with a fresh installation?

Otherwise, considering no one else has been able to reproduce this, I'd argue it would be good to close it (assuming this is an underlying problem of environment with matlplotib or another dep)

**Comment by RBMindee** [comment]
Actually I tried again one or two weeks ago and it does not occur anymore... I have no idea what changed... It was on a very fresh linux installation, so I'm randomly guessing some package were not updated ? Anyway, although it is a bit frustrating, I guess we can close the PR as the initial script does not reproduce the problem on my machine.

**Comment by fg-mindee** [comment]
Good to hear @RBMindee :+1: 
Most likely it was a setup issue with one of the deps then!
